portfolio/src/main/webapp/welcome_data.js

```diff
@@ -0,0 +1,9 @@
+function getWelcomeText(){
+    console.log("trying to get welcome text");
+    fetch('/welcome-data')
+    .catch(console.log('failed to get resposne'))
+    .then(response => response.text())
+    .catch(console.log('failed to convert to text'))
+    .then(welcomeText => document.getElementById('welcome-data').innerText=welcomeText)
+    .catch(console.log('failed to print text to html element'));
```

Comment:
Nice that you're catching individual points of failure separately to make things easier to debug!